DSharpPlus.Core/Rest/RatelimitBucket.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace DSharpPlus.Core.Rest
+{
+
+    // represents a single ratelimit bucket, against a single endpoint.
+    internal record RatelimitBucket
+    {
+        private const string UnlimitedRatelimitIdentifier = "unlimited";
+        private const int DefaultLimit = 1;
```

Comment:
90% of endpoints are 5, actually, but I can understand wanting to only allow 1 request.